bin/saya/scripts/3_piltover.sh

```
@@ -0,0 +1,77 @@
+
+# Display the transaction hash and contract address for reference
+echo "Piltover deploy transaction hash: $TRANSACTION_HASH"
+echo "Piltover contract address: $PILTOVER_CONTRACT_ADDRESS"
+
+# Function to check the finality status of the transaction on L2
+check_finality() {
+    local tx_hash="$1"    # Transaction hash to check the status for
+    local url="$2"        # Starknet RPC URL
+
+    # Use sncast to retrieve the transaction status
+    local result=$(sncast --account dev tx-status "$tx_hash" --url "$url")
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)